### DISCUSSION TITLE
Expected output of Debug tab in and tests in Console tab in cfclient.

### DISCUSSION THREAD

**Discussion by varadVaidya** [bitcraze/discussions#1671] [Q&A] [answered]
Hello,
I am currently facing some increased crash rate while performing autonomous flights, using cflib scripts and crazyswarm2. To find out the root  cause of these issues, i want to remove any hardware discrepancies. One of the checks that i found in the discussions forums is the debug tab in `Help -> About -> Debug`. 

<details open>
<summary>This is my current output:</summary>
<br>

```
Cfclient
Cfclient version: 2024.11
System: linux
Python: 3.10.12
Qt: 6.7.1
PyQt: 6.7.1

Interface status
radio: Crazyradio version 99.55
UsbCdc: No information available
udp: None
prrt: No information available
cpx: None

Input readers
linuxjsdev (1 devices connected)

Input devices
linuxjsdev: (0) Microsoft X-Box 360 pad

Crazyflie
Connected: radio://0/80/2M
Firmware: 7c64c98060db (CLEAN)

Decks found
bcFlow2: rev=A, adr=0D29C00001000091

Sensors found
BMP3XX: 1
imuPhi: 0.0
imuTheta: 0.0
imuPsi: 0.0
AK8963: 0
LPS25H: 0

Sensors tests
MPU6500: 0
AK8963: 0
LPS25H: 0
imuPhi: 0.0
imuTheta: 0.0
imuPsi: 0.0
```

</details>

This is the console log, with all the test in the cfclient performed, except for the propeller test as I am facing the same issue as in [#1430 ](https://github.com/bitcraze/crazyflie-firmware/issues/1430)

<details open>
<summary>Console Log with Tests</summary>
<br>

```
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 9:7c64c98060db (2024.10.2 +9) CLEAN
SYS: I am 0x38353530353947090036002C and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcFlow2 for deck 0
ZR2: Z-down sensor [OK]
PMW: Motion chip id: 0x49:0xB6
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.10 (CF21)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 15048 bytes
STAB: Starting stabilizer loop
ESTKALMAN: State out of bounds, resetting
SUP: Can not fly
SUP: Ready to fly
SYS: No assert information found
SYSLOAD: Task dump
SYSLOAD: Load	Stack left	Name
SYSLOAD: 3.77 	231 	Tmr Svc
SYSLOAD: 62.70 	54 	IDLE
SYSLOAD: 0.0 	48 	PWRMGNT
SYSLOAD: 1.93 	212 	CRTP-RX
SYSLOAD: 2.12 	75 	ZRANGER2
SYSLOAD: 0.94 	117 	RATE_SUPE
SYSLOAD: 2.43 	189 	FLOW
SYSLOAD: 0.0 	254 	CMDHL
SYSLOAD: 0.0 	265 	PASSTHROU
SYSLOAD: 0.0 	116 	PLATFORM-
SYSLOAD: 0.0 	223 	MEM
SYSLOAD: 0.0 	244 	LOG
SYSLOAD: 0.04 	257 	PARAM
SYSLOAD: 0.34 	158 	SYSTEM
SYSLOAD: 0.07 	62 	CRTP-TX
SYSLOAD: 5.20 	346 	STABILIZE
SYSLOAD: 7.09 	144 	SENSORS
SYSLOAD: 6.23 	315 	KALMAN
SYSLOAD: 2.62 	175 	SYSLINK
SYSLOAD: 3.27 	91 	LEDSEQCMD
SYSLOAD: 1.17 	108 	CRTP-SRV
SYSLOAD: 0.0 	114 	USBLINK
HEALTH: Idle:3.96V sag: 0.53V (< 0.70V) [OK]
ESTKALMAN: WARNING: Kalman prediction rate off (96)
STAB: WARNING: stabilizer loop rate is off (947)
STORAGE: Used storage: 5 item stored, 127 Bytes/7168 Bytes (1%)
STORAGE: Fragmentation: 0%
STORAGE: Efficiency: Data: 5 Bytes (0%), Keys: 107 Bytes (1%), Metadata: 15 Bytes (0%)
```

</details>

This log seems to be more or less same across multiple crazyflies. 

I just want to confirm with the team that hardware of my current crazyflies is fine. My suspicion with the hardware comes from similar behaviour as seen in [#1406](https://github.com/bitcraze/crazyflie-firmware/issues/1406). The gyro plots used to get stuck at a particular value, with crazyflie outputting 99% thrust on all motors. This doesn't seem to be the case anymore on the said drone, but is enough for me to be suspicious for other hardware problems.

The self tests seem to pass (as seen by the green light flashing during startup). Although some crazyflies don't show any green light during startup, but this was also the behaviour before the sudden crash rate increase (with some flights facing issues similar to [#1424](https://github.com/bitcraze/crazyflie-firmware/issues/1424)).

If there are any more tests that can confirm the hardware healthiness, please let me know.

Thanks for all the help!

**Comment by gemenerik** [comment]
That gyro does sound suspicious. The changes in https://github.com/bitcraze/crazyflie-firmware/pull/1415 added functionality to kill the motors after a timeout if the gyro fails. If you've updated the firmware since then, it might explain why you are not seeing the fixed thrust anymore, though you still should see similar behavior for a brief period before the timeout kicks in. @tobbeanton might be able to suggest how to measure the interrupt if you have access to an oscilloscope, or share other ideas for troubleshooting.

**Reply by varadVaidya** [reply]
Yes. I am running on the latest main branch of the crazyflie-firmware, particularly to solve the gyro problems as the mentioned PR fixes by adding a supervisor. I do have access to oscilloscope, and would love to measure the interrupt you mentioned. For now i think i might have boiled down at least some of the crashes to some form of wrong state estimation in kalman filter. Few new crazyflie are also on the way, to compare against brand new hardware.

I would also love if you could confirm if the debug tests output look good. The sensor tests all showing 0 seems confusing to me.

**Reply by tobbeanton** [reply, marked as answer]
A lot is pointing towards an IMU problem. If you have access to reflow soldering tools I would reflow the BMI088 which most likely will fix the problem.

**Comment by varadVaidya** [comment]
Closing this as an IMU problem, we got new crazyflies and everything is working just fine. Still haven't got around to reflow the IMU, but i think thats the problem
